sklearn/preprocessing/data.py

```diff
@@ -1719,10 +1726,15 @@ def _transform_selected(X, transform, selected="all", copy=True):
         X_sel = transform(X[:, ind[sel]])
         X_not_sel = X[:, ind[not_sel]]
 
-        if sparse.issparse(X_sel) or sparse.issparse(X_not_sel):
-            return sparse.hstack((X_sel, X_not_sel))
+        if order:
+            # As of now, X is expected to be dense array
+            X[:, ind[sel]] = X_sel
+            return X
         else:
```

Comment:
Nit: One of my pet peeves is inconsequential `else` statements. You can remove this `else` statements without any changes in your logic. (It reduces one level of indentations.)